osp/views/answer.py

```diff
@@ -0,0 +1,79 @@
+from rest_framework import viewsets, status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from osp.models import (
+    Question, Answer, ChoiceValue, CheckboxValue, DropdownValue, DateValue,
+    TimeValue, FileUploadValue, ShortAnswerValue, ParagraphValue, FormFeedback
+)
+from osp.serializers.answer import AnswerSerializer
+from osp.serializers.form_feedback import FormFeedbackSerializer
+from osp.utils.answer_function import get_model_and_serializer
+
+class AnswerView(viewsets.ModelViewSet):
+
+    serializer_class = AnswerSerializer
+    permission_classes = [IsAuthenticated]
+    queryset = Answer.objects.all()
+
+    def get_queryset(self):
+        queryset = self.queryset.order_by('question__order')
+        form_id = self.request.query_params.get('form_id', None)
+        question_id = self.request.query_params.get('question_id', None)
+        user = self.request.user
+        if form_id is not None:
+            queryset = queryset.filter(form_answers__form_id=form_id)
+        if user is not None:
+            queryset = queryset.filter(form_answers__user_id=user.id)
+        if question_id is not None:
+            queryset = queryset.filter(question_id=question_id)
+        
+        # response
+        return queryset
+
+    # GET request
+    def list(self, request):
+        objs = self.get_queryset()
+        results = []
+
+        for obj in objs:
+            value = get_model_and_serializer(obj.question.data_type)
+            model = value['model']
+            serializer = value['serializer']
+            instance = model.objects.get(id=obj.id)
+            results.append(serializer(instance).data)
+
+        # response
+        return Response(results, status=status.HTTP_200_OK)
+
+    # POST request
+    def create(self, request):
+        user = self.request.user
+        results = []
+        answers = []
+
+        for obj in request.data['answers']:
+            data_type = Question.objects.get(id=obj['question']).data_type
+            value = get_model_and_serializer(data_type)
+            model = value['model']
+            serializer = value['serializer']
+            id = obj.get('id', None)
+            if id is not None:
+                instance = model.objects.get(id=id)
+                instance = serializer(instance, data=obj)
+                instance.is_valid(raise_exception=True)
+                instance.save(question_id=obj['question'])
+            else:
+                instance = serializer(data=obj)
+                instance.is_valid(raise_exception=True)
+                instance.save(question_id=obj['question'])
```

Comment:
no... never do this, never pass anything inside save() it causes issues later on